packages/api/src/controllers/access-control.ts

```diff
@@ -50,73 +57,184 @@ app.post(
+          );
+        }
+
+        const signingKey = signingKeyOutput[0];
+
+        if (signingKey.userId !== content.userId) {
+          console.log(`
+            access-control: gate: disallowing playback for contentId=${content.id} with playbackId=${playbackId} the content and the public key pub=${req.body.pub} do not share the same owner, signingKeyUserId=${signingKey.userId}, contentUserId=${content.userId}
+          `);
+          throw new NotFoundError("Content not found");
+        }
+
+        if (signingKey.disabled || signingKey.deleted) {
+          const signingKeyLog = JSON.stringify(JSON.stringify(signingKey));
+          console.log(`
+            access-control: gate: disallowing playback for contentId=${content.id} with playbackId=${playbackId} the public key pub=${signingKey.id} is disabled or deleted, signingKey=${signingKeyLog}
+          `);
+          throw new ForbiddenError("The public key is disabled or deleted");
+        }
+
+        tracking.recordSigningKeyValidation(signingKey.id);
+        res.set("Cache-Control", "max-age=120,stale-while-revalidate=600");
+        res.status(204);
+        return res.end();
+      case "webhook":
+        if (!req.body.accessKey) {
+          console.log(`
+            access-control: gate: no accessKey provided for playbackId ${playbackId}, disallowing playback
+          `);
+          throw new ForbiddenError(
+            "Content is gated and requires an access key"
+          );
+        }
+        const webhook = await db.webhook.get(content.playbackPolicy.webhookId);
+        if (!webhook) {
+          console.log(`
+            access-control: gate: content with playbackId ${playbackId} is gated but corresponding webhook not found for webhookId ${content.playbackPolicy.webhookId}, disallowing playback
+          `);
+          throw new ForbiddenError(
+            "Content is gated and corresponding webhook not found"
+          );
+        }
+        const statusCode = await fireGateWebhook(
+          webhook,
+          content.playbackPolicy,
+          req.body.accessKey
         );
-      }
-
-      if (signingKeyOutput.length > 1) {
-        let collisionKeys = JSON.stringify(signingKeyOutput);
+        switch (statusCode) {
+          case 200:
+            res.set("Cache-Control", "max-age=120,stale-while-revalidate=600");
+            res.status(204);
+            return res.end();
+          case 0:
+            console.log(`
+              access-control: gate: content with playbackId ${playbackId} is gated but webhook ${webhook.id} failed, disallowing playback
+            `);
+            throw new BadGatewayError(
+              "Content is gated and corresponding webhook failed"
+            );
+          default:
+            console.log(`
+              access-control: gate: content with playbackId ${playbackId} is gated but webhook ${webhook.id} returned status code ${statusCode}, disallowing playback
```

Comment:
Can you change all of these logs to be structured? Loki is not able to parse "free form text" like this and it is troublesome to make any use out of them on grafana. We use something similar to this format: https://brandur.org/logfmt

e.g. This could be `access-control: gate: content gated but webhook returned bad status code, disallowing playback playbackId=${playbackId} webhookId=${webhook.id} statusCode=${statusCode}`
This is only 1 example, all logs (at least the new ones) should be structured

Reply:
Right! Makes sense, will make these changes and comment out or remove most of the logs to be less verbose